logprep/processor/processor_registry.py

```diff
@@ -58,4 +61,9 @@ def get_processor_class(cls, processor_type):
         _type_
             _description_
         """
+        if processor_type == "delete":
+            warnings.warn(
+                "delete processor is deprecated and will be removed in a future release. Please use deleter instead.",
+                DeprecationWarning,
+            )
```

Comment:
Does this warning actually work? When I tried implementing the first deprecation the warning was never thrown, so I used the regular logging. Maybe you can write a test to check if it actually works.

Reply:
I found this bit in the [Python Docs](https://docs.python.org/3/library/warnings.html#warning-categories):

DeprecationWarning:
> Base category for warnings about deprecated features when those warnings are intended for other Python developers (ignored by default, unless triggered by code in __main__).

This might explain why deprecation warnings sometimes do not show up.
I now changed the warning type to `UserWarning` which should definitely show up and also added a test.

UserWarnings also make sense in this case, since it's mainly users of Logprep we want to warn about the breaking change.

Reply:
Okay thanks, that's good to know. I will try this out with the next deprecation I'll write.